### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by rakshi00**
Add push notification on platform
Add toast for password change
Add possibility to reset purchase (withdraw) even when volunteer has already confirmed

### CODE REVIEW

File: frontend/admin/src/pages/home/PasswordChange.tsx [hunk 1]
```diff
@@ -37,12 +39,16 @@ const PasswordChange = () => {
             oldPassword,
             newPassword
         })
-            .then(() => history.push("/"))
+            .then(() => {
+                history.push("/")
+                toast('Passwort ändern', 'Passwort wurde geändert.');
```
Comment (ntruessel) [inline frontend/admin/src/pages/home/PasswordChange.tsx:43]: I'd switch the order. `history.push()` suggests that you leave this page so you should not perform any more actions after leaving.

File: frontend/admin/src/pages/purchase/PurchaseDetail.tsx [hunk 2]
```diff
@@ -416,7 +434,7 @@ const PurchaseDetailInternal = (props: any) => {
                               id="internalComments"/>
                 </div>
                 <div className="form-group">
-                    <label htmlFor="displayTableOrderItems">Einkaufsliste</label>
+                    <label htmlFor="displayTableOrderItems"><i>Die Einkaufsliste kann nicht mehr verändert oder gepeichert werden, nachdem der Helfer oder die Helferin den Einkauf abgeschlossen hat.</i></label>
```
Comment (ntruessel) [inline frontend/admin/src/pages/purchase/PurchaseDetail.tsx]: You should not be able to edit after accepting not after completion (there might be receipts already)

File: backend/src/main/java/de/njsm/versusvirus/backend/service/purchase/PurchaseService.java [hunk 3]
```diff
@@ -159,6 +156,15 @@ public void withdrawPurchase(UUID purchaseId) {
                 purchase.getVolunteerApplications().clear();
             }
             purchase.setStatus(NEW);
+        } else if (purchase.getStatus() == VOLUNTEER_ACCEPTED) {
+            telegramApi.deleteMessage(groupChatId, purchase.getBroadcastMessageId());
```
Comment (ntruessel) [inline backend/src/main/java/de/njsm/versusvirus/backend/service/purchase/PurchaseService.java:164]: This is unsafe. There might be submitted receipts for this purchase, potentially screwing with business logic further in the purchase process.

**Review by ntruessel — CHANGES_REQUESTED** [review]

File: backend/src/main/java/de/njsm/versusvirus/backend/service/purchase/PurchaseService.java (hunk 3 above)
Comment (rakshi00) [reply backend/src/main/java/de/njsm/versusvirus/backend/service/purchase/PurchaseService.java:164]: True, you are correct. I forgot that we are still in this state after the receipts come in but before all receipts have been received. I'll correct that

Comment (rakshi00) [reply backend/src/main/java/de/njsm/versusvirus/backend/service/purchase/PurchaseService.java:164]: @ntruessel Fixed, please have a look :) https://github.com/ElsaHilftDigital/main/pull/127/commits/67d95fca30c7e80ece3129bbcaff04f993681150

File: frontend/admin/src/pages/purchase/PurchaseDetail.tsx (hunk 2 above)
Comment (rakshi00) [reply frontend/admin/src/pages/purchase/PurchaseDetail.tsx]: Removed the input field and trash cans completely after VOLUNTEER_ACCEPTED

**Review by ntruessel — APPROVED** [review]

**Pull request merged**